### PR TITLE
Pr branch

### PR DESCRIPTION
**Pull request by Atharva-77**
This is the code for review for PR. Do let me know

### CODE REVIEW

File: to-do-list/src/App.js [hunk 1]
```diff
@@ -1,29 +1,73 @@
...
-                      <Route path="/form" element={<Form />} />
-                      <Route path="/view-employees" element={<ViewEmp />} />
-                      <Route path="/" element={<MainDataPg />} />
-              </Routes>
-        </BrowserRouter>
-        
+       
+           
+        <ColorContext.Provider value= {{empDetails,setempDetails,allId,setallId}}>
+              <BrowserRouter >
```
Comment (chandreshpatidar) [inline to-do-list/src/App.js]: Move Router code in new component `Router.jsx`

File: to-do-list/src/App.js [hunk 2]
```diff
@@ -1,29 +1,73 @@
+// This is a PR branch.
+// I have added comments to this file
+
+      
+//Pages & compo...
+//git merge
+//git conflict
+//git amendment
+
+
```
Comment (chandreshpatidar) [inline to-do-list/src/App.js]: Remove these lines

File: to-do-list/src/App.js [hunk 3]
```diff
@@ -1,29 +1,73 @@
...
+import React,{useState} from 'react';
```
Comment (chandreshpatidar) [inline to-do-list/src/App.js]: Add VS code extensions `Pretier, `

File: to-do-list/src/App.js [hunk 4]
```diff
@@ -1,29 +1,73 @@
...
 import logo from './logo.svg';
 import './App.css';
-import MainPg from './MainPg';
+import MainPg from './Pages/Unused-Pages/MainPg';
```
Comment (chandreshpatidar) [inline to-do-list/src/App.js]: Remove un-used components from entire repo

File: to-do-list/src/App.js [hunk 5]
```diff
@@ -1,29 +1,73 @@
...
 import {BrowserRouter, Routes, Route, Link} from 'react-router-dom';
-import Form from './Form';
-import MainDataPg from './MainDataPg';
-import ViewEmp from './ViewEmp';
+import Form from './Components/Form';
+import MainDataPg from './Pages/MainDataPg';
```
Comment (chandreshpatidar) [inline to-do-list/src/App.js]: Keep meaningful name

File: to-do-list/src/App.js [hunk 6]
```diff
@@ -1,29 +1,73 @@
...
+import ViewEmp from './Pages/ViewEmp';
+import CallForm from './Pages/Unused-Pages/CallForm';
+
+import { ColorContext } from './Components/Context'
```
Comment (chandreshpatidar) [inline to-do-list/src/App.js]: Move context into `services -> context -> filename`

File: to-do-list/src/App.js [hunk 7]
```diff
@@ -1,29 +1,73 @@
...
+import Heading from './Components/Heading';
 
 function App() {
+
```
Comment (chandreshpatidar) [inline to-do-list/src/App.js]: Remove redundent blank line

File: to-do-list/src/App.js [hunk 8]
```diff
@@ -1,29 +1,73 @@
...
-import ViewEmp from './ViewEmp';
+import Form from './Components/Form';
+import MainDataPg from './Pages/MainDataPg';
+import ViewEmp from './Pages/ViewEmp';
+import CallForm from './Pages/Unused-Pages/CallForm';
+
+import { ColorContext } from './Components/Context'
+import Heading from './Components/Heading';
 
 function App() {
```
Comment (chandreshpatidar) [inline to-do-list/src/App.js]: Use Arrow funtcion syntax

File: to-do-list/src/App.js [hunk 9]
```diff
@@ -1,29 +1,73 @@
...
+
+  const [empDetails,setempDetails]=useState([])
```
Comment (chandreshpatidar) [inline to-do-list/src/App.js]: `setempDetails` reformat in case-case

File: to-do-list/src/App.js [hunk 10]
```diff
@@ -1,29 +1,73 @@
...
-                      <Route path="/form/:id" element={<Form />} />
-                      <Route path="/form" element={<Form />} />
-                      <Route path="/view-employees" element={<ViewEmp />} />
-                      <Route path="/" element={<MainDataPg />} />
-              </Routes>
-        </BrowserRouter>
-        
+       
+           
+        <ColorContext.Provider value= {{empDetails,setempDetails,allId,setallId}}>
```
Comment (chandreshpatidar) [inline to-do-list/src/App.js]: Rename `ColorContext`

File: to-do-list/src/App.js [hunk 11]
```diff
@@ -1,29 +1,73 @@
...
+                                                } />
+
+                            <Route path="/" element= 
+                                                {
+                                                      <>
+                                                            <Heading />
+                                                           <MainDataPg />
+                                                      </>
+                                                } />
+                         </Routes>
```
Comment (chandreshpatidar) [inline to-do-list/src/App.js]: Reformat with prettier

File: to-do-list/src/App.js [hunk 12]
```diff
@@ -1,29 +1,73 @@
...
+       
+           
+        <ColorContext.Provider value= {{empDetails,setempDetails,allId,setallId}}>
+              <BrowserRouter >
+                        <Routes>
+                   
+                            <Route path="/update-employee/:id" element= 
+                                                {
+                                                      <>
+                                                            <Heading />
```
Comment (chandreshpatidar) [inline to-do-list/src/App.js]: Move ` <Heading />` outside of routes

File: to-do-list/src/Components/Context.js [hunk 13]
```diff
@@ -0,0 +1,3 @@
+import React,{useState} from 'react';
```
Comment (chandreshpatidar) [inline to-do-list/src/Components/Context.js]: Move this file in services folder as explain in App.js file

Remove un-used variable.,
Import `createContext` as module

File: to-do-list/src/App.css [hunk 14]
```diff
@@ -1,34 +1,32 @@
 :root
 {
-    --red: red;
+    --red: #8b5cf6;
     --dark-green: green;
-    --light-lessOpacity-green: rgb(186, 253, 87,0.8);
-    --light-moreOpacity-green: rgb(175, 245, 71);
+    --light-lessOpacity-green:#8ff5ffed;
```
Comment (chandreshpatidar) [inline to-do-list/src/App.css]: Check this color code

File: to-do-list/src/App.css [hunk 15]
```diff
@@ -1,34 +1,32 @@
 :root
```
Comment (chandreshpatidar) [inline to-do-list/src/App.css]: Format all files with prettier
Remove un-used css

File: to-do-list/src/App.css [hunk 16]
```diff
@@ -1,34 +1,32 @@
...
+
+    --light-moreOpacity-green:#64cfff;
 }
 .App {
   text-align: center;
-  /* background-color: aqua; */
 }
 
-.Heading_div
+.heading_Div
```
Comment (chandreshpatidar) [inline to-do-list/src/App.css]: `heading_Div` -> `headingDiv`

File: to-do-list/src/App.css [hunk 17]
```diff
@@ -1,34 +1,32 @@
...
 {
     margin-top: 20px;
-    /* font-size: larger; */
```
Comment (chandreshpatidar) [inline to-do-list/src/App.css:16]: Remove commented code

File: to-do-list/src/App.css [hunk 18]
```diff
@@ -1,34 +1,32 @@
...
     color: var(--red);
     font-weight: 600;
     text-decoration: none;
-    /* background-color: aqua; */
 }
 
 
-.Heading_div:hover
+.heading_Div:hover
 {
```
Comment (chandreshpatidar) [inline to-do-list/src/App.css]: Remove redundant empty line

File: to-do-list/src/Components/Form.css [hunk 19]
```diff
@@ -10,28 +10,30 @@
     border: 1px solid;
    
 }
```
Comment (chandreshpatidar) [inline to-do-list/src/Components/Form.css]: Add blank line between two classnames

File: to-do-list/src/Components/Form.js [hunk 20]
```diff
@@ -0,0 +1,240 @@
+import React,{useContext, useEffect, useState} from 'react';
+import './Form.css'
```
Comment (chandreshpatidar) [inline to-do-list/src/Components/Form.js]: Import css files at bottom/top of all import

File: to-do-list/src/Components/Form.js [hunk 21]
```diff
@@ -0,0 +1,240 @@
+import React,{useContext, useEffect, useState} from 'react';
+import './Form.css'
+import { useParams, useLocation } from 'react-router';
+
```
Comment (chandreshpatidar) [inline to-do-list/src/Components/Form.js]: Remove blank line

File: to-do-list/src/Components/Form.js [hunk 22]
```diff
@@ -0,0 +1,240 @@
...
+import './Form.css'
+import { useParams, useLocation } from 'react-router';
+
+import { useNavigate } from 'react-router-dom';
+import { ColorContext } from './Context'
+import InputComponent from './InputComponent';
+//only 1 func
+//e.t.value
+
+//new compo for input fie;d
```
Comment (chandreshpatidar) [inline to-do-list/src/Components/Form.js]: Remove these lines

File: to-do-list/src/Components/Form.js [hunk 23]
```diff
@@ -0,0 +1,240 @@
...
+    //     intial_Firstname=idOfEmp.firstName;
+    //     initial_Lastname=idOfEmp.lastName;
+    //     initial_empid=idOfEmp.empid;
+    //     initial_age=idOfEmp.age;
+
+    //     button_value='Update Details'
+
+    //     console.log("VALues",finalEmp.firstName);
+    // }
+
```
Comment (chandreshpatidar) [inline to-do-list/src/Components/Form.js]: Remove commented code

File: to-do-list/src/Components/Form.js [hunk 24]
```diff
@@ -0,0 +1,240 @@
...
+            {idpresent && <span>{message}</span>}
+
+            <div className='input_Main_Container'>
+                
+                <div className='input_Container'> 
+                    <InputComponent className_Prop='inputTag' spanName="FirstName" value_Prop={firstName} onChange_Prop={e=>firstNameFunc(e)} placeholder='Enter First Name'/>
+                </div>
+
+                <div className='input_Container'>
+                    <InputComponent className_Prop='inputTag' spanName="LastName" value_Prop={lastName} onChange_Prop={e=>lastNameFunc(e)} placeholder='Enter Last Name'/>
```
Comment (chandreshpatidar) [inline to-do-list/src/Components/Form.js]: e=>lastNameFunc(e) -> `lastNameFunc`

**Review by chandreshpatidar — CHANGES_REQUESTED** [review]

**Comment by Atharva-77** [comment]
Closing this PR. Resolved the issues

**Pull request closed without merging**